### ISSUE TITLE
Watcher errors out on multidirectory downloads (Needs recursive scanning)

### ISSUE THREAD

**Issue by ywLBhGk2noZLbY7** [nosmokingbandit/watcher#15]
Sometimes downloads are unpacked and you are left with multiple directories filled with a large amount of bloat. For example, here is a slightly modified directory tree of an nzb watcher downloaded:

```
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/3t90Dv4guDsdM9J-JR19vZs307eMxxP/
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/3t90Dv4guDsdM9J-JR19vZs307eMxxP/TFTF/
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/3t90Dv4guDsdM9J-JR19vZs307eMxxP/TFTF/flyer.jpg
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/3t90Dv4guDsdM9J-JR19vZs307eMxxP/abc.mkv
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/3t90Dv4guDsdM9J-JR19vZs307eMxxP/pic.jpg
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/3t90Dv4guDsdM9J-JR19vZs307eMxxP/text.url
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/INFO/
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/INFO/INFO.txt
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/INFO/INFO.URL
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/INFO/INFO.jpg
Jz2O2Q4kwjHgJQB-VM8g3SkQUhqCE8s/INFO/INFO.nfo
```

In these instances, [os.listdir()](https://github.com/nosmokingbandit/watcher/blob/master/core/postprocessing.py#L132) is not recursive and causes watcher to error out with the following sort of error:

> INFO 2016-12-25 00:12:43,667 core.postprocessing.complete: Setting MOVIE status.
> INFO 2016-12-25 00:12:43,667 core.sqldb.get_distinct: Getting distinct values for tt0000000 in SEARCHRESULTS
> INFO 2016-12-25 00:12:43,668 core.updatestatus.movie_status: Setting MOVIES tt0000000 status to Finished.
> INFO 2016-12-25 00:12:43,668 core.sqldb.update: Updating tt0000000 to Finished in MOVIES.
> INFO 2016-12-25 00:12:43,710 core.sqldb.update: Updating tt0000000 to 2016-12-25 in MOVIES.
> INFO 2016-12-25 00:12:43,766 core.postprocessing.renamer: Renaming  to Ubuntu (1970)
> ERROR 2016-12-25 00:12:43,776 core.postprocessing.renamer: Renamer failed: Could not rename file.
> Traceback (most recent call last):
>   File "/opt/watcher/app/core/postprocessing.py", line 519, in renamer
>     os.rename(abs_path_old, abs_path_new)
> OSError: [Errno 22] Invalid argument
> INFO 2016-12-25 00:12:43,777 core.postprocessing.mover: Moving /downloads/completed/Movies/Ubuntu Directory/ to /movies/Ubuntu (1970)

**Comment by nosmokingbandit** [comment]
Fixed in commit https://github.com/nosmokingbandit/watcher/commit/45900da8e532b3093ddaf2ff1aeff712a25341bc

**Comment by ywLBhGk2noZLbY7** [comment]
That commit, I think, breaks the fix made in 2a84f90b660dc42c3e21d6c0bedb62782ef9396f. I'm on my phone so I can't test it, but I think it is now causing the issue described in #27.

**Comment by nosmokingbandit** [comment]
You are correct. Git got those lines all screwed up.

It should be fixed in *again* in https://github.com/nosmokingbandit/watcher/commit/1c869c227ea13d0e7c2f6bdae97aa1412a6cd036.

I changed the way the string is formatted so hopefully these lines will stay how I have them from now on.
